metricbeat/module/system/cpu/cpu.go

```diff
@@ -0,0 +1,50 @@
+package cpu
+
+import (
+	"github.com/elastic/beats/metricbeat/helper"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	_ "github.com/elastic/beats/metricbeat/module/system"
+	topbeat "github.com/elastic/beats/topbeat/beater"
+)
+
+func init() {
+	helper.Registry.AddMetricSeter("system", "cpu", MetricSeter{})
+}
+
+type MetricSeter struct{}
+
+func (m MetricSeter) Setup() error {
+	return nil
+}
+
+func (m MetricSeter) Fetch(ms *helper.MetricSet) (events []common.MapStr, err error) {
+
+	cpuStat, err := topbeat.GetCpuTimes()
+	if err != nil {
+		logp.Warn("Getting cpu times: %v", err)
+		return nil, err
+	}
+
+	event := common.MapStr{
+		"user":     cpuStat.User,
+		"system":   cpuStat.Sys,
+		"nice":     cpuStat.Nice,
+		"idle":     cpuStat.Idle,
+		"iowait":   cpuStat.Wait,
+		"irq":      cpuStat.Irq,
+		"softirq":  cpuStat.SoftIrq,
+		"steal":    cpuStat.Stolen,
+		"user_p":   cpuStat.UserPercent,
+		"system_p": cpuStat.SystemPercent,
+	}
```

Comment:
Could we somehow create the event (MapStr) in the topbeat code? Otherwise we have to remember adding the key here as well every time we add something to topbeat.